File: docs/api/logout/logout.py
import requests, sys

server = 'http://127.0.0.1:3000'
user = 'trackhub1'
password = 'trackhub1'

def login(server, user, password):
    r = requests.get(server+'/api/login', auth=(user, password), verify=False)
    if not r.ok:
        print "Couldn't login, reason: %s [%d]" % (r.text, r.status_code)
        sys.exit

    auth_token = r.json()[u'auth_token']
    print 'Logged in [%s]' % auth_token
    return auth_token

auth_token = login(server, user, password)
r = requests.get(server+'/api/logout', headers={ 'user': user, 'auth_token': auth_token })
if not r.ok:
    print "Couldn't logout, reason: %s [%d]" % (r.text, r.status_code)
    sys.exit
print 'Logged out'

File: docs/api/logout/logout.rb
require 'net/https'
require 'uri'

require 'rubygems'
require 'json'

def login(user, pass)
  request = Net::HTTP::Get.new('/api/login')
  request.basic_auth(user, pass)
  response = $http.request(request)
  
  if response.code != "200"
    puts "Couldn't login, reason: #{response.body} [#{response.code}]"
    exit
  end

  result = JSON.parse(response.body)
  puts "Logged in [#{result["auth_token"]}]"
  
  return result["auth_token"]
end

server = 'http://127.0.0.1:3000'
user = 'trackhub1'
pass = 'trackhub1'

url = URI.parse(server)
$http = Net::HTTP.new(url.host, url.port)
# $http.use_ssl = true
# $http.verify_mode = OpenSSL::SSL::VERIFY_NONE

auth_token = login(user, pass)

request = Net::HTTP::Get.new('/api/logout')
request["User"] = user
request["Auth-Token"] = auth_token
response = $http.request(request)
 
if response.code != "200"
  puts "Invalid response: #{response.code}"
  puts response.body
  exit
end

puts 'Logged out'
